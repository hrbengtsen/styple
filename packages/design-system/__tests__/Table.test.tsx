import * as React from "react";
import { render, RenderResult } from "@testing-library/react";
import { Table, Thead, Tbody, Tfoot, Tr, Th, Td } from "..";

describe("Test Table", () => {
  it("should render as expected", () => {
    const view: RenderResult = render(
      <Table
        css={{
          maxWidth: "$4xl",
        }}
      >
        <Thead>
          <Tr>
            <Th>Prop</Th>
          </Tr>
        </Thead>
        <Tbody>
          <Tr>
            <Th>defaultOpen</Th>
          </Tr>
        </Tbody>
        <Tfoot>
          <Tr>
            <Th>modal</Th>
          </Tr>
        </Tfoot>
      </Table>
    );

    expect(view).toMatchSnapshot();
  });
});
