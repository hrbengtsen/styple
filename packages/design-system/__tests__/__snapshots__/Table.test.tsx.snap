// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Table should render as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="c-hMqJpz c-hMqJpz-ebinYA-direction-horizontal c-hMqJpz-dMdkyz-cv c-hMqJpz-lbtSuu-cv"
      >
        <div
          class="c-lesPJm c-lesPJm-idWksIc-css"
        >
          <table
            class="c-eyJUct c-eyJUct-iaQqDl-css"
          >
            <thead
              class="c-gsnRPq"
            >
              <tr
                class="c-PJLV"
              >
                <th
                  class="c-dLElCH c-dLElCH-heqqZn-align-start"
                >
                  Prop
                </th>
              </tr>
            </thead>
            <tbody
              class="c-fGHEql"
            >
              <tr
                class="c-PJLV"
              >
                <th
                  class="c-dLElCH c-dLElCH-heqqZn-align-start"
                >
                  defaultOpen
                </th>
              </tr>
            </tbody>
            <tfoot
              class="c-PJLV"
            >
              <tr
                class="c-PJLV"
              >
                <th
                  class="c-dLElCH c-dLElCH-heqqZn-align-start"
                >
                  modal
                </th>
              </tr>
            </tfoot>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="c-hMqJpz c-hMqJpz-ebinYA-direction-horizontal c-hMqJpz-dMdkyz-cv c-hMqJpz-lbtSuu-cv"
    >
      <div
        class="c-lesPJm c-lesPJm-idWksIc-css"
      >
        <table
          class="c-eyJUct c-eyJUct-iaQqDl-css"
        >
          <thead
            class="c-gsnRPq"
          >
            <tr
              class="c-PJLV"
            >
              <th
                class="c-dLElCH c-dLElCH-heqqZn-align-start"
              >
                Prop
              </th>
            </tr>
          </thead>
          <tbody
            class="c-fGHEql"
          >
            <tr
              class="c-PJLV"
            >
              <th
                class="c-dLElCH c-dLElCH-heqqZn-align-start"
              >
                defaultOpen
              </th>
            </tr>
          </tbody>
          <tfoot
            class="c-PJLV"
          >
            <tr
              class="c-PJLV"
            >
              <th
                class="c-dLElCH c-dLElCH-heqqZn-align-start"
              >
                modal
              </th>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
